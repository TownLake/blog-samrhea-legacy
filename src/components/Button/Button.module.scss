@import "src/assets/scss/variables";
@import "src/assets/scss/mixins";

.button {
  border: 1px solid var(--color-button-border);
  border-radius: $button-border-radius;
  color: var(--color-button-color);
  display: inline-block;
  font-size: $typographic-base-font-size;
  font-weight: normal;
  height: $button-height;
  line-height: $button-height;
  margin-left: auto;
  margin-right: auto;
  text-align: center;
  text-decoration: none;
  @include padding(0, 1);

  &:hover,
  &:focus {
    color: $color-blue;
  }
}
