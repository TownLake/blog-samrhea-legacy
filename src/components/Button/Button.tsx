import React from "react";

import cn from "classnames";
import { Link } from "gatsby";

import * as styles from "./Button.module.scss";

interface Props {
  className?: string;
  title: string;
  to: string;
}

const Button: React.FC<Props> = ({ className, title, to }: Props) => (
  <Link className={cn(styles.button, className)} to={to}>
    {title}
  </Link>
);

export default Button;
