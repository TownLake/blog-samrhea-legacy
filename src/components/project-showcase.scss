.project-showcase {
    max-width: 1200px;
    margin: 0 auto;
    padding: 2rem;
  
    h1 {
      text-align: center;
      margin-bottom: 2rem;
      font-size: 2rem;
      font-weight: bold;
    }
  
    .project-grid {
      display: grid;
      grid-template-columns: repeat(auto-fill, minmax(300px, 1fr));
      gap: 2rem;
    }
  }
  
  // Dark mode styles
  @media (prefers-color-scheme: dark) {
    .project-showcase {
      background-color: #1a202c;
      color: #f7fafc;
    }
  }