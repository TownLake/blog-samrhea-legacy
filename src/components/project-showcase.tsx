.project-tile {
    position: relative;
    width: 100%;
    height: 0;
    padding-bottom: 75%; // 4:3 aspect ratio
    overflow: hidden;
    border-radius: 8px;
    box-shadow: 0 4px 6px rgba(0, 0, 0, 0.1);
    transition: transform 0.3s ease;
  
    &:hover {
      transform: translateY(-5px);
    }
  
    .project-link {
      text-decoration: none;
      color: inherit;
    }
  
    .project-image {
      position: absolute;
      top: 0;
      left: 0;
      width: 100%;
      height: 100%;
      background-size: cover;
      background-position: center;
    }
  
    .project-overlay {
      position: absolute;
      bottom: 0;
      left: 0;
      right: 0;
      background: rgba(0, 0, 0, 0.7);
      color: white;
      padding: 1rem;
      transform: translateY(100%);
      transition: transform 0.3s ease;
    }
  
    &:hover .project-overlay {
      transform: translateY(0);
    }
  
    .project-title {
      margin: 0 0 0.5rem;
      font-size: 1.2rem;
    }
  
    .project-description {
      margin: 0;
      font-size: 0.9rem;
      opacity: 0.8;
    }
  }