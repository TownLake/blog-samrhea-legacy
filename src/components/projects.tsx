import React from 'react';
import { Link } from 'gatsby';
import './project.scss';

interface ProjectProps {
  title: string;
  description: string;
  imageUrl: string;
  projectUrl: string;
}

const Project: React.FC<ProjectProps> = ({ title, description, imageUrl, projectUrl }) => {
  return (
    <div className="project-tile">
      <Link to={projectUrl} className="project-link">
        <div className="project-image" style={{ backgroundImage: `url(${imageUrl})` }}>
          <div className="project-overlay">
            <h3 className="project-title">{title}</h3>
            <p className="project-description">{description}</p>
          </div>
        </div>
      </Link>
    </div>
  );
};

export default Project;