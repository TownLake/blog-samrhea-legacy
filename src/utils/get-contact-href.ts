import { btoa } from "abab";

const getContactHref = (name: string, contact: string) => {
  const hrefs: { [key: string]: string } = {
    email: btoa(contact) || "",
    line: `line://ti/p/${contact}`,
    telegram: `https://t.me/${contact}`,
    vkontakte: `https://vk.com/${contact}`,
    medium: `https://medium.com/${contact}`,
    github: `https://github.com/${contact}`,
    weibo: `https://www.weibo.com/${contact}`,
    gitlab: `https://www.gitlab.com/${contact}`,
    codepen: `https://www.codepen.io/${contact}`,
    twitter: `https://www.twitter.com/${contact}`,
    soundcloud: `https://soundcloud.com/${contact}`,
    facebook: `https://www.facebook.com/${contact}`,
    instagram: `https://www.instagram.com/${contact}`,
    linkedin: `https://www.linkedin.com/in/${contact}`,
    youtube: `https://www.youtube.com/channel/${contact}`,
    mastodon: `${contact}`,
  };

  return hrefs[name] ?? contact;
};

export default getContactHref;
