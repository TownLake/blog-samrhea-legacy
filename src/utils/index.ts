export { default as getIcon } from "./get-icon";
export { default as getContactHref } from "./get-contact-href";
export { default as toKebabCase } from "./to-kebab-case";
export { default as getDefaultColorMode } from "./get-default-color-mode";
export * as testUtils from "./test-utils";
