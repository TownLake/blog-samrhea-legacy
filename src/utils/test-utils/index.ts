export {
  default as renderWithCoilProvider,
  createSnapshotsRenderer,
} from "./render-with-coil-provider";
export { default as getMeta } from "./get-meta";
