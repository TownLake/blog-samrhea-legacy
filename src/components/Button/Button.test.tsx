import React from "react";

import {
  render as reactTestingLibraryRender,
  screen,
} from "@testing-library/react";

import { Button } from "@/components/Button";
import { createSnapshotsRenderer } from "@/utils/test-utils";

describe("Button", () => {
  test("renders correctly", () => {
    const props = { title: "Button", to: "/" };
    const tree = createSnapshotsRenderer(<Button {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("renders correctly with external link", () => {
    const props = { title: "Button", to: "https://example.com" };
    const tree = createSnapshotsRenderer(<Button {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("renders correctly with custom class name", () => {
    const props = { title: "Button", to: "/", className: "custom-class-name" };
    const tree = createSnapshotsRenderer(<Button {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("the title is rendered correctly and exists", () => {
    const props = { title: "Text", to: "/" };
    reactTestingLibraryRender(<Button {...props} />);
    expect(screen.getByText(props.title)).toBeInTheDocument();
  });
});
