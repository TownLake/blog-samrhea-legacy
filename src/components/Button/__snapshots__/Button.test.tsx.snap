// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly 1`] = `
<a
  className=""
  href="/"
>
  Button
</a>
`;

exports[`Button renders correctly with custom class name 1`] = `
<a
  className="custom-class-name"
  href="/"
>
  Button
</a>
`;

exports[`Button renders correctly with external link 1`] = `
<a
  className=""
  href="https://example.com"
>
  Button
</a>
`;
